From a73b828f67893e5795950f19185b657980ec5d89 Mon Sep 17 00:00:00 2001
From: Bogdan Hamciuc <bogdan.hamciuc@nxp.com>
Date: Fri, 6 Dec 2019 11:32:52 +0200
Subject: [PATCH 035/269] s32g: Fix DFS initialization routine

Take individual ports out of reset before checking for lock status.

Upstream-Status: Pending 

Signed-off-by: Bogdan Hamciuc <bogdan.hamciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
Issue: ALB-4275
---
 plat/s32g/s32g_clocks.c  | 1 +
 plat/s32g/s32g_pinctrl.c | 1 -
 2 files changed, 1 insertion(+), 1 deletion(-)

diff --git a/plat/s32g/s32g_clocks.c b/plat/s32g/s32g_clocks.c
index 39fb18849..8c5fa9205 100644
--- a/plat/s32g/s32g_clocks.c
+++ b/plat/s32g/s32g_clocks.c
@@ -179,6 +179,7 @@ static void enable_dfs(enum s32g_dfs_type dfs,
 	/* deassert individual ports */
 	for (port = 0; port < S32G_DFS_PORTS_NR; port++)
 		reset |= (params[port][DFS_PORT_EN] << port);
+	mmio_write_32(DFS_PORTRESET(dfs), ~reset);
 
 	/* wait until all configured ports are locked */
 	while ((mmio_read_32(DFS_PORTSR(dfs)) & reset) != reset)
diff --git a/plat/s32g/s32g_pinctrl.c b/plat/s32g/s32g_pinctrl.c
index 8d1e3a31c..66103db3d 100644
--- a/plat/s32g/s32g_pinctrl.c
+++ b/plat/s32g/s32g_pinctrl.c
@@ -10,7 +10,6 @@
 
 static void linflex_config_pinctrl(int lf)
 {
-	asm volatile ("b	.");
 	assert(lf == 0);
 
 	/* set PC09 - MSCR[41] - for UART0 TXD */
-- 
2.17.1

